### PR TITLE
Docker update

### PR DESCRIPTION
**Pull request by matthewford**
## 📝 A short description of the changes

* Update the docker image to run the application both locally and for production
* Added compose files for RAILS_ENV development and production
* Update the README
* Fixed an outdated URL as part of testing developing within Docker

## 🔗 Link to the relevant story (or stories)

* https://app.asana.com/0/1207904458299124/1207285373295115

## :shipit: Deployment implications

* This should not impact production as we're still using buildpacks on the PaaS; this is in preparation for the migration of the PaaS.

## ✅ Checklist

- [x] Features that cannot go live are behind a feature flag/env var or specify deploy date and open PR as draft 
- [x] I have checked that commit messages make sense and explain the reasoning for each change
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have squashed any unnecessary or part-finished commits

### CODE REVIEW

File: init-hstore.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+CREATE EXTENSION IF NOT EXISTS hstore;
```
Comment (dreamfall) [inline init-hstore.sql:1]: ✂️ no newline

File: config/database.yml [hunk 2]
```diff
@@ -1,24 +1,17 @@
...
   <<: *default
-  <<: *credentials
-  database: qae_test
+  url: <%= ENV['DATABASE_URL'] %>
 
 production:
   <<: *default
   url: <%= PaasResolver.pgsql_uri %>
-  pool: <%= ENV['DB_POOL'] || ENV['MAX_THREADS'] || 5 %>
+  pool: <%= ENV['DB_POOL'] || ENV['MAX_THREADS'] || 5 %>
```
Comment (dreamfall) [inline config/database.yml:17]: ✂️ no newline

File: bin/docker-entrypoint [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash -e
+
+# If running the rails server then create or migrate existing database
+if [ "${1}" == "./bin/rails" ] && [ "${2}" == "server" ]; then
+  /rails/bin/rails db:prepare
+fi
+
+exec "${@}"
```
Comment (dreamfall) [inline bin/docker-entrypoint:8]: ✂️ no newline

File: nginx.conf [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+        }
+
+        location /files {
+            internal;
+            alias /rails/public;
+        }
+    }
+}
```
Comment (dreamfall) [inline nginx.conf:66]: ✂️ no newline

**Review by Lubosky — APPROVED** [review]

**Pull request merged**